lib/actions/ssh.ts

```diff
@@ -295,27 +290,32 @@ export const ssh: CommandDefinition<
 					for the \`ssh\` requests.
 
 					Attempting the unproxied request for now.`);
-				return '';
+				return;
 			}
 
-			const p = proxyConfig!;
-			const tunnelOptions: Dictionary<string> = {
-				proxy: `${p.host}:${p.port}`,
-				dest: '%h:%p',
-			};
+			const p = proxyConfig;
 			if (p.username && p.password) {
-				tunnelOptions.user = p.username;
-				tunnelOptions.pass = p.password;
+				// proxytunnel understands these variables for proxy authentication.
+				// Setting the variables instead of command-line options avoids the
+				// need for shell-specific escaping of special characters like '$'.
+				process.env.PROXYUSER = p.username;
+				process.env.PROXYPASS = p.password;
 			}
 
-			const ProxyCommand = `proxytunnel ${bash.args(tunnelOptions, '--', '=')}`;
-			return `-o ${bash.args({ ProxyCommand }, '', '=')}`;
+			return [
+				'proxytunnel',
+				`--proxy=${p.host}:${p.port}`,
+				// ssh replaces these %h:%p variables in the ProxyCommand option
+				// https://linux.die.net/man/5/ssh_config
+				'--dest=%h:%p',
+				...(verbose ? ['--verbose'] : []),
+			];
```

Comment:
Modified to return an array of unescaped strings instead of a bash-escaped string, which is platform centric and does not work with PowerShell / cmd.exe. By using the PROXYUSER and PROXYPASS env vars set above, it was possible to avoid shell escaping and quoting altogether. To be fair, `proxytunnel` is not available on PowerShell / cmd.exe anyway, but being shell-agnostic is a good thing. This also allowed removing the `bash` dependency from `package.json`.